velox/exec/HashAggregation.cpp

```diff
@@ -410,10 +410,18 @@ RowVectorPtr HashAggregation::getOutput() {
     return nullptr;
   }
 
+  const auto& queryConfig = operatorCtx_->driverCtx()->queryConfig();
   if (isDistinct_) {
     if (!newDistincts_) {
       if (noMoreInput_) {
         finished_ = true;
+        if (groupingSet_->hasDefaultGlobalGroupingSetOutput()) {
+          prepareOutput(queryConfig.preferredOutputBatchRows());
```

Comment:
We only need 1 or 2 rows, no? Why create full-size vector?

Reply:
Yeah, we need 1 or 2 rows.

The original AggregationNode we have the metadata from is released at this point and so I'll have to expose additional APIs from GroupingSet for it. Have added that.